### PR TITLE
Inventory Plot dialog autofills defined elements, autofill implemented for multiple receivers, Is_Element property created for climatic dataframes

### LINKED ISSUE #5929: Small change in Climatic > Check Data > Inventory
In this dialogue I usually include all the elements in the file.  Most of them are defined as climatic elements in the Define > Climatic dialogue.  Please could the default be to include them by default in the dialogue.

### PR DESCRIPTION
**Pull request by dannyparsons**
fixes #5929
This replaces #5951 (not branched)

- Autofill enabled for multiple receivers - all valid columns will autofill (unlike for single receivers where autofill only happens if there is exactly one valid column)
- `Is_Element` metadata property added for climatic data frames. Receivers can be set to autofill all climatic elements with `Receiver.SetClimaticType("element")`
- Implemented for the elements receiver on Inventory Plot dialog

@africanmathsinitiative/developers This is ready for review.

### CODE REVIEW

File: instat/ucrReceiver.vb [hunk 1]
```diff
@@ -524,7 +525,24 @@ Public Class ucrReceiver
 
     Public Sub SetClimaticType(strTemp As String)
         Dim dctTemp As New Dictionary(Of String, String())
-        dctTemp.Add("Climatic_Type", {Chr(34) & strTemp & Chr(34)})
+
+        ' "element" is a special case since it is a different metadata property.
+        If strTemp = "element" Then
+            dctTemp.Add("Is_Element", {"TRUE"})
+            SetIncludedAutoFillProperties(dctTemp)
```
Comment (dannyparsons) [inline instat/ucrReceiver.vb:532]: This allows developers to set this property in the same way i.e. `ucrReceiverElements.SetClimaticType("element")` although it's a different metadata property to specific elements.

File: instat/ucrReceiverSingle.vb [hunk 2]
```diff
@@ -307,10 +307,6 @@ Public Class ucrReceiverSingle
         RemoveSelected()
     End Sub
 
-    Private Sub Selector_DataFrameChanged() Handles ucrSelector.DataFrameChanged
-        CheckAutoFill()
-    End Sub
```
Comment (dannyparsons) [inline instat/ucrReceiverSingle.vb:312]: Moved to base class as multiple receiver now does the same.

File: instat/ucrReceiverSingle.vb [hunk 3]
```diff
@@ -319,11 +315,6 @@ Public Class ucrReceiverSingle
         End If
     End Sub
 
-    Protected Overrides Sub Selector_ResetAll()
-        MyBase.Selector_ResetAll()
-        CheckAutoFill()
-    End Sub
```
Comment (dannyparsons) [inline instat/ucrReceiverSingle.vb:325]: Moved to base class as multiple receiver now does the same.

**Comment by dannyparsons** [comment]
Note that this will not work with existing `DataBook` files which are already defined as climatic, without defining as climatic again, since the `Is_Element` property won't be in the metadata for those files.

**Review by rdstern — CHANGES_REQUESTED** [review]
@dannyparsons I think this dialogue may be working, but it has re-uncovered another bug.

I tried with 6 stations from Zambia, defined as climatic (can check with the metadata) and it doesn't autofill in this dialogue.  I pressed reset and still nothing.

I returned to the define climatic which filled automatically and then returned to the dialogue.  Now it works - at least when I press reset.

I think Rachel found this issue when preparing the videos., i.e. that the information on the climatic definition doesn't completely survive saving.  It does in the metadata, but not completely.  I hope this message is clear.

**Comment by dannyparsons** [comment]
Ok, yes that sounds like a general issue with the climatic metadata. I suggest not addressing that in this PR.

If there's not an issue for this already can we make one for it?

**Review by lloyddewit — APPROVED** [review]

**Review by shadrackkibet — APPROVED** [review]

**Comment by lloyddewit** [comment]
@rdstern Please could you approve? thanks

**Review by rdstern — APPROVED** [review]

**Pull request merged**